Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -1233,11 +1235,16 @@ extension SwiftLanguageServer {
         let hints = infos
           .lazy
           .filter { !$0.hasExplicitType }
-          .map { info in
-            InlayHint(
-              position: info.range.upperBound,
-              category: .type,
-              label: info.printedType
+          .map { info -> InlayHint in
+            let position = info.range.upperBound
+            let label = ": \(info.printedType)"
```

Comment:
It would be really nice if we could use `InlayHintLabelPart`s here and make each label part clickable and to jump to that type’s definition. But that’s definitely out-of-scope for this PR and would probably also require changes to `sourcekitd`.

Reply:
I agree, that would be a cool feature for a future PR.